src/main/java/zmaster587/advancedRocketry/tile/multiblock/machine/TileChemicalReactor.java

```diff
@@ -109,12 +107,26 @@ public void consumeItemsSpecial(IRecipe recipe) {
 			for (IInventory hatch : itemInPorts) {
 				for (int i = 0; i < hatch.getSizeInventory(); i++) {
 					ItemStack stackInSlot = hatch.getStackInSlot(i);
-					for (ItemStack stack : ingredient) {
-						if (!stackInSlot.isEmpty() && stackInSlot.getCount() >= stack.getCount() && (stackInSlot.getItem() == stack.getItem() && (stackInSlot.getItemDamage() == stack.getItemDamage() || stack.getItemDamage() == OreDictionary.WILDCARD_VALUE))) {
+					for(ItemStack stack : ingredient) {
+						if(!stackInSlot.isEmpty() && stackInSlot.getCount() >= stack.getCount() && (stackInSlot.getItem() == stack.getItem() && (stackInSlot.getItemDamage() == stack.getItemDamage() || stack.getItemDamage() == OreDictionary.WILDCARD_VALUE))) {
 							ItemStack stack2 = hatch.decrStackSize(i, stack.getCount());
 
-							if (stack2.getItem() instanceof ItemArmor) {
-								stack2.addEnchantment(AdvancedRocketryAPI.enchantmentSpaceProtection, 1);
+							if(stack2.getItem() instanceof ItemArmor) {
+								boolean alreadyHasEnchantment = false;
+
+								NBTTagList enchantmentList = stack2.getEnchantmentTagList();
+								for(int j = 0 ; j < enchantmentList.tagCount(); j++) {
+									NBTTagCompound tag = (NBTTagCompound)enchantmentList.get(j);
+									if(tag.getInteger("id") == Enchantment.getEnchantmentID(AdvancedRocketryAPI.enchantmentSpaceProtection)) {
+										alreadyHasEnchantment = true;
+										break;
+									}
+								}
```

Comment:
Cache enchantment ID before loop. Enchantment ID won't change during loop. Also you can use `EnchantmentHelper.getEnchantmentLevel(enchantment: Enchantment, stack: ItemStack): Int` instead probably but it's up to you

Reply:
Ah, I'll see if I can replace the whole loop with that, thanks!